adam/curriculum/generate_segmentation_results.py

```diff
@@ -0,0 +1,196 @@
+import argparse
+import base64
+import json
+import logging
+import shutil
+from pathlib import Path
+import random
+from typing import Any, Mapping
+
+import cv2
+import numpy as np
+
+import requests
+import yaml
+from tqdm import tqdm
+
+API_ROUTE = "/instanceseg"
+DESCRIPTION_FILENAME = "description.yaml"
+INFO_FILENAME = "info.yaml"
+COLOR_SEED = 42
+LABEL_COLORS = np.random.uniform(0, 255, size=(50, 3))
+STEGO_MODEL = "stego"
+MASK_RCNN_MODEL = "rcnn"
+
+
+def stego_postprocessing(data: Mapping[str, Any], img: Any) -> Any:
+    # Comment so current unused data in the `data` map can be recalled in the future
+    # boxes, label = data["boxes"], data["labels"]
+    masks = np.array(data["masks"])
+
+    beta = 0.6
+    alpha = 1.0
+    gamma = 0.0
+    for mask in masks:
+        r_map = np.zeros_like(mask).astype(np.uint8)
+        g_map = np.zeros_like(mask).astype(np.uint8)
+        b_map = np.zeros_like(mask).astype(np.uint8)
+
+        # mask color assignment
+        color = LABEL_COLORS[random.randrange(0, len(LABEL_COLORS))]
+        r_map[mask == 1], g_map[mask == 1], b_map[mask == 1] = color
+
+        # combine masks
+        segmentation_map = np.stack([r_map, g_map, b_map], axis=2)
+        img = np.array(img)
+        img = cv2.cvtColor(img, cv2.COLOR_RGB2BGR)
+
+        # instance mask
+        cv2.addWeighted(img, alpha, segmentation_map, beta, gamma, img)
+
+    return img
```

Comment:
Ah, I meant to ask if you thought the stroke extraction code would better handle the output from this post-processing method or the RCNN post-processing? Either way I'll probably leave both in here (with a comment explaining the unused function) in case we discover a need to modify the output of this for stroke extraction. I'd like to easily have the reference available.